src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -561,6 +561,13 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             if (repeatCount==0) {
                 return this;
             }
+
+            // this is where we can check if the repeatCount will put us over m_MaxCapacity
+            // We are doing the check here to prevent the corruption of the StringBuilder.
+            if (Length + repeatCount > m_MaxCapacity || repeatCount + Length < repeatCount) {
```

Comment:
Nit: It would be nice to cache `Length + repeatCount` into a local to make this easier to read, and also to help the JIT to generate better code for it.